### PR TITLE
JS: model string functions from closure library

### PR DESCRIPTION
**Pull request by asger-semmle**
Adds models for the functions in [goog.string](https://google.github.io/closure-library/api/goog.string.html) from the closure standard library.

### CODE REVIEW

File: javascript/ql/src/semmle/javascript/frameworks/ClosureLibrary.qll [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+          name = "trimRight" or
+          name = "unescapeEntities" or
+          name = "urlDecode" or
+          name = "urlEncode" or
+          name = "whitespaceEscape"
+        )
+        or
+        (name = "truncate" or name = "truncateMiddle") and
+        pred = getArgument(0) and
+        not getArgument(1).getIntValue() < 8 // length of <script>
```
Comment (esben-semmle) [inline javascript/ql/src/semmle/javascript/frameworks/ClosureLibrary.qll]: this seems very specific to a (insufficient) XSS sanitizer, is this conjunct meant to be right here? Wouldn't it be better to have as a barrier/sanitizer?

File: javascript/ql/src/semmle/javascript/frameworks/ClosureLibrary.qll [hunk 2]
```diff
@@ -0,0 +1,58 @@
+/**
+ * Provides models for miscellaneous utility functions in the closure standard library.
```
Comment (esben-semmle) [inline javascript/ql/src/semmle/javascript/frameworks/ClosureLibrary.qll]:
```suggestion
 * Provides models for functions in the closure standard library.
```
(we have no other occurrences of "miscellaneous" in our docstrings)

Comment (asger-semmle) [reply javascript/ql/src/semmle/javascript/frameworks/ClosureLibrary.qll]: But there are many functions in the closure library that aren't modelled here. They are in concept-specific libraries like `UriLibraries.qll`, so this really is just the category for things that don't fit anywhere else.

File: javascript/ql/src/semmle/javascript/frameworks/ClosureLibrary.qll (hunk 1 above)
Comment (asger-semmle) [reply javascript/ql/src/semmle/javascript/frameworks/ClosureLibrary.qll]: What's the difference?

Comment (esben-semmle) [reply javascript/ql/src/semmle/javascript/frameworks/ClosureLibrary.qll]: You can easily get a good command injection in less than 8 characters, so the barrier should not be used for `js/command-injection`, for instance.

Comment (asger-semmle) [reply javascript/ql/src/semmle/javascript/frameworks/ClosureLibrary.qll]: Good point. I've just changed them to regular taint steps for now; the length check wasn't motivated by any real code anyway.

**Comment by xiemaisi** [comment]
LGTM as well and looks harmless enough to not need an evaluation. Merging.

**Pull request merged**